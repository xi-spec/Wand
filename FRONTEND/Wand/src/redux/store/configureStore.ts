import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
