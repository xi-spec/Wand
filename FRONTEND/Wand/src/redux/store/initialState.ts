import { InitialState } from '../../models/index';

const initialState:InitialState = {
  products: {
    allProducts: [],
    productsList: [],
    filterProducts: {
      foundList: [],
      notFound: false
    },

    selectedProduct: {
      _id: '',
      name: '',
      description: '',
      motto: '',
      cost: {
        value: 0,
        currency: ''
      },
      images: {
        alls: [
          ''

        ],
        cover: ''
      },

      weight: {
        value: 0,
        volum: ''
      },
      material: '',
      Packaging: '',
      measurements: '',
      type: {
        name: '',
        image: ''
      },
      stock: 0
    }
  },
  productsType: [],
  cart: {
    cartList: [],
    totalCost: 0
  },
  user: {
    _id: '',
    name: '',
    email: '',
    photoURL: '',
    phone: 0,
    creationDate: '',
    password: '',
    address: {
      street: '',
      postcode: '',
      country: '',
      city: ''

    },
    card: {
      cardNumber: '',
      securtyCode: '',
      cardholder: '',
      expiryDate: ''
    },
    favourite: [],
    orderHistory: [],
    isLogged: false,
    isExist: false,
    loginError: false,
    isRegisted: false

  }
};

export default initialState;
