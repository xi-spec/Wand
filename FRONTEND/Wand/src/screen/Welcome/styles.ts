import { StyleSheet } from 'react-native';
import { SIZE, COLORS, FONTS } from '../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  image: {
    flex: 1,
    resizeMode: 'cover',
    width: '100%',
    height: '100%'

  },
  intoduction: {
    flex: 2,
    paddingLeft: 100,
    paddingTop: 330
  },
  text: {
    color: 'white',
    fontSize: 50,
    fontFamily: FONTS.logo

  },
  textSmall: {
    paddingTop: 20,
    marginRight: 50,
    color: 'white',
    fontFamily: FONTS.primaryMedium,
    fontSize: SIZE.buttonSize

  },
  buttonContainer: {
    flex: 3,
    paddingTop: 50,
    alignItems: 'center'

  },
  button: {
    width: 250,
    height: 50,
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 20

  },
  blueButton: {
    backgroundColor: COLORS.primary
  },
  whiteButton: {
    borderColor: 'white',
    borderWidth: 2,
    borderStyle: 'solid'
  },
  buttonText: {
    color: 'white',
    fontSize: SIZE.buttonSize

  },
  laterText: {
    color: 'white',
    fontSize: 15,
    textDecorationLine: 'underline'
  }
});

export default styles;
