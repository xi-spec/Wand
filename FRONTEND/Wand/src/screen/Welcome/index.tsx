import React from 'react';
import {
  ImageBackground,
  View,
  Text,
  TouchableOpacity
} from 'react-native';

import { IMAGES } from '../../constants/images';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import { Params } from '../../models';
import styles from './styles';

export default function Welcome ({ navigation }:{ navigation: NavigationScreenProp<NavigationRoute<Params>, Params>}) {
  return (
<View style={styles.container}>
    <ImageBackground style={styles.image}source={{ uri: IMAGES.login }}>
        <View style={styles.intoduction}>
        <Text style={styles.text}>Wand</Text>
        <Text style={styles.textSmall}>Three young asian girls in Spain founded WAND in 2020 for design and jewelry, which is the short form </Text>
        <Text style={styles.textSmall}>  “What A Nice Day”.</Text>
        </View>

        <View style={styles.buttonContainer}>
        <TouchableOpacity
         testID='login'
    style={{ ...styles.button, ...styles.blueButton }}
    onPress={() =>
      navigation.navigate('Login')
    }
      >
        <Text style={styles.buttonText}>LOGIN</Text>
      </TouchableOpacity>
        <TouchableOpacity
         testID='signUp'
    style={{ ...styles.button, ...styles.whiteButton }}
    onPress={() =>
      navigation.navigate('Register')
    }
      >
        <Text style={styles.buttonText}>SIGNUP</Text>
      </TouchableOpacity>

   <TouchableOpacity
    onPress={() =>
      navigation.navigate('Home')}
      testID='home'
   >

      <Text style={styles.laterText}>Maybe later</Text>
   </TouchableOpacity>
        </View>

    </ImageBackground>
    </View>
  );
}
