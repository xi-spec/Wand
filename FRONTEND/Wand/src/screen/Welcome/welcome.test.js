
import React from 'react';
import Welcome from './index';
import { render, fireEvent, act } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import store from '../../redux/store/configureStore';

describe('Given a Welcome component', () => {
  let navigation;

  beforeEach(() => {
    navigation = {
      navigate: jest.fn()
    };
  });
  describe('When is invoked', () => {
    test('Then it is renders correctly', () => {
      const tree = render(<Provider store={store}><Welcome /></Provider>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test('Then has 1 child', () => {
      const tree = render(<Provider store={store}><Welcome /></Provider>).toJSON();
      expect(tree.children.length).toBe(1);
    });
  });

  describe('When is press in testId login', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Welcome navigation={navigation} /></Provider>);
      const touch = getByTestId('login');

      act(() => {
        fireEvent.press(touch);
      });

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
  describe('When is press in testId signUp', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getByTestId } = render(<Welcome navigation={navigation}/>);
      const touch = getByTestId('signUp');

      act(() => {
        fireEvent.press(touch);
      });

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When is press in testId home', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getByTestId } = render(<Welcome navigation={navigation}/>);
      const touch = getByTestId('home');

      act(() => {
        fireEvent.press(touch);
      });

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
