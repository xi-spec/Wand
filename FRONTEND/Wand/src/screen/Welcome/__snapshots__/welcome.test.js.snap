// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Welcome component When is invoked Then it is renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://trello-attachments.s3.amazonaws.com/603be1b68ce9c654d30ae543/6045f407c8709551d44f44e2/87f38de716a88d59cfe99595f6a9bd49/Olive_Klein_Amber_Agate_8.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 2,
          "paddingLeft": 100,
          "paddingTop": 330,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "RalewaySemiBold",
            "fontSize": 50,
          }
        }
      >
        Wand
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratMedium",
            "fontSize": 20,
            "marginRight": 50,
            "paddingTop": 20,
          }
        }
      >
        Three young asian girls in Spain founded WAND in 2020 for design and jewelry, which is the short form 
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "MontserratMedium",
            "fontSize": 20,
            "marginRight": 50,
            "paddingTop": 20,
          }
        }
      >
          “What A Nice Day”.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "paddingTop": 50,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "opacity": 1,
            "width": 250,
          }
        }
        testID="login"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          LOGIN
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "white",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "opacity": 1,
            "width": 250,
          }
        }
        testID="signUp"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          SIGNUP
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="home"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "textDecorationLine": "underline",
            }
          }
        >
          Maybe later
        </Text>
      </View>
    </View>
  </View>
</View>
`;
